utils/demangle.c

```diff
@@ -1469,6 +1470,11 @@ static int dd_source_name(struct demangle_data *dd)
 	if (add_name)
 		dd_append_len(dd, p, num);
 	__dd_consume_n(dd, num, NULL);
+	separator = strstr(dd->new, "..");
```

Comment:
I'm afraid it might have multiple `".."`.  Does this handle it properly?

Reply:
I'm not sure why, but in my test case, it seems to work properly. But to make sure, I'll add a loop for that.

Reply:
I think it worked by luck.  It just converts a single instance, but later `dd_source_name()` call would convert it again.  If the original name contains more than two ".." and it has a `dd_source_name()` after that, not all ".." would be converted.

Reply:
You're right. Thanks for the review.